src/customnodesession.cpp

```diff
@@ -196,35 +205,23 @@ Status CustomNodeSession::createBlob(const struct CustomNodeTensor* tensor, Infe
             error.str());
         return StatusCode::NODE_LIBRARY_INVALID_CONTENT_SIZE;
     }
-    auto allocator = std::make_shared<CustomNodeOutputAllocator>(*tensor, library, customNodeLibraryInternalManager);
-    // auto allocator = std::make_shared<CustomNodeOutputAllocator_2>(*tensor, library, customNodeLibraryInternalManager);
+    auto allocatorImpl = std::make_shared<CustomNodeOutputAllocator_2>(*tensor, library, customNodeLibraryInternalManager);
+    auto allocator = ov::runtime::Allocator(allocatorImpl);
     try {
         switch (tensor->precision) {
         case CustomNodeTensorPrecision::FP32:
-            resultBlob = InferenceEngine::make_shared_blob<float>(desc, std::move(allocator));
-            break;
         case CustomNodeTensorPrecision::I32:
-            resultBlob = InferenceEngine::make_shared_blob<int32_t>(desc, std::move(allocator));
-            break;
         case CustomNodeTensorPrecision::I8:
-            resultBlob = InferenceEngine::make_shared_blob<int8_t>(desc, std::move(allocator));
-            break;
         case CustomNodeTensorPrecision::U8:
-            resultBlob = InferenceEngine::make_shared_blob<uint8_t>(desc, std::move(allocator));
-            break;
         case CustomNodeTensorPrecision::FP16:
-            resultBlob = InferenceEngine::make_shared_blob<uint16_t>(desc, std::move(allocator));
-            break;
         case CustomNodeTensorPrecision::I16:
-            resultBlob = InferenceEngine::make_shared_blob<int16_t>(desc, std::move(allocator));
-            break;
         case CustomNodeTensorPrecision::U16:
-            resultBlob = InferenceEngine::make_shared_blob<uint16_t>(desc, std::move(allocator));
+            resultBlob = std::make_shared<ov::runtime::Tensor>(ov::element::Type(ovmsPrecisionToIE2Precision(toInferenceEnginePrecision(tensor->precision))), ov::Shape(shape), allocator);
             break;
         case CustomNodeTensorPrecision::UNSPECIFIED:
             return StatusCode::INTERNAL_ERROR;
         }
-    } catch (const InferenceEngine::Exception& e) {
+    } catch (const InferenceEngine::Exception& e) {  // TODO change exception type
```

Comment:
Why not now?

Reply:
Already done - newest change was not pushed to remote.